challenge/files/11_maps.toml

```diff
@@ -0,0 +1,141 @@
+name = "Maps"
+category = "easy"
+default_code_snippet = """
+func main() {
+	var m map[int]bool
+	fmt.Println(m)
+
+	val, ok := m[1]
+	fmt.Println(val, " ", ok)
+
+	for i := 0; i < 5; i++ {
+		m[i] = i%2 == 0
+	}
+
+	val, ok = m[1]
+	fmt.Println(val, " ", ok)
+
+	fmt.Println(m)
+
+	var toPrint []int
+	for key := range m {
+		toPrint = append(toPrint, key)
+	}
+
+	fmt.Println(toPrint)
+
+	delete(m, 2)
+	fmt.Println(m)
+
+	clear(m)
+	fmt.Println(m)
+}
+"""
+learning_advise = """
+“Learning is never done without error, and defeat ” – Vladimir Lenin.
+"""
+
+[[learning_links]]
+title = "Go blog about maps"
+url = "https://go.dev/blog/maps"
+
+[[learning_links]]
+title = "Source code"
+url = "https://go.dev/src/runtime/map.go"
+
+[[learning_links]]
+title = "Go by examples: Maps"
+url = "https://gobyexample.com/maps"
+
+[[questions]]
+text = """Select the true statements and the line they refer to.
+"""
+type = "select_answers"
+answers = [
+    { text = "The map can be used without initialization" },
```

Comment:
I think that first question overwhelms. We need to split it.

<img width="972" alt="Screenshot 2024-08-11 at 15 47 57" src="https://github.com/user-attachments/assets/3ff92e5e-d85a-472f-8f4b-7ba6d4a48b42">

### First question.

```go
func main() {
	var m map[int]bool
	
        val, ok := m[1]
	fmt.Println(val, " ", ok)
	
        for i := 0; i < 5; i++ {
		m[i] = i%2 == 0
	}

	fmt.Println(m)
```

What program prints - not compiles, panic, false false, false false and panic

### Fixed initialization questions

Replace code with new snippet with fixed map initialization. 

```go
func main() {
	var m map[int]bool
	fmt.Println(m)
	
        for i := 0; i < 5; i++ {
		m[i] = i%2 == 0
	}

	fmt.Println(m)
```

What program prints. Plus next question - why.

### random order question

```go
func main() {
	var m map[int]bool
	fmt.Println(m)

	val, ok := m[1]
	fmt.Println(val, ok)

	for i := 0; i < 5; i++ {
		m[i] = i%2 == 0
	}

	var toPrint []int
	for key := range m {
		toPrint = append(toPrint, key)
	}

	fmt.Println(toPrint)
}
```

What it prints. Next question - why

### clean question

I think we should skip it.

Reply:
You can change co with question `new_code_snippet` field

<img width="491" alt="Screenshot 2024-08-11 at 16 36 47" src="https://github.com/user-attachments/assets/bf424e65-66db-44f7-8662-0528919a0598">